docs/configuration.md

```diff
@@ -181,6 +181,31 @@ REMS uses [Logback](https://logback.qos.ch/) for logging. By default everything
 
     java -Dlogback.configurationFile=logback-prod.xml -jar rems.jar run
 
+### Extended Logging
+
+The config option `:extended-logging` is for routes that should additionally log the content of entity mutations.
+
+Logging is at `INFO` level and the log messages are prefixed with `extended-logging (:uri request)` e.g. `extended-logging /api/forms/edit`.
```

Comment:
Should the logging happen to the audit_log in DB, to the application log, or a separate audit log file (potentially replacing audit log in DB)?

Reply:
wrt our previous conversation on issue #3184, I got the impression that the audit log and its functionality wasn't a widely used feature. 
Furthermore, for our needs, it is a requirement that services relay logs to a separate logging service. In which case, using application logs simplifies integration greatly, as such it would be ideal if they could provide an authoritative source for audits.
I would imagine that other organisations will have similar requirements. But I have no basis for that statement other than a hunch.

Reply:
Audit log as a feature is used, i.e. the logs go to the database, in the REMS production instances. It's just that the use is such that not many features have been built on top of it (like browsing in the UI). Also nobody else has requested to log the request / response, so it has not been implemented.

It is just a forward-looking idea to implement an option to have audit log as a separate file, instead of the DB. To me it sounds like you would want the whole audit log as a clean file, and not bother with the DB table at all. And possibly it doesn't make sense to log everything always to the regular log file. There is going to be a lot to optimize in the logs if a compact log file is to be stored for a long time.

We would normally work out a rough design for the idea, that would include potential features to be implemented first and some left as optional. Not all REMS production instances are interested in this logging, so a simple set of features to choose from would be nice.

We can usually accept a partial feature easier if it's useful, a part of the whole set and by default is not configured to be used everywhere. Not every (improvement) idea that is discovered in a review typically is implemented. It could be a follow-up task or an idea for later. Just something to think about (the whole).

Reply:
That makes sense.

I reckon it might be useful in the future, if users want the audit log to be able to provide an authoritative source on the state of an entity at a given time.

Though you could also argue that the audit log is for high-level auditing, but if you want to investigate a specific incident, then you would also need to go over the application logs. At that point, it would probably bloat the audit log beyond its initial use-case.

For our specific use case, with this feature implemented, we are able to comply with all our logging requirements (and effectively our auditability requirements), by monitoring the application log and routing relevant log lines to a separate logging service. So, the audit log will probably not see a lot of use, whether it is a db or a clean file.